src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -41,36 +36,70 @@ export const LegacyLogin = () => {
     handleSubmit,
     register,
     formState: { errors, isSubmitting },
+    reset,
   } = useForm({ shouldUnregister: true })
 
   const translate = useTranslate()
 
-  const isLoginError = (err: any): err is LoginError => err && typeof err.message === 'string'
+  const isLoginError = (err: any): err is LoginError =>
+    err && typeof err.response?.data?.error?.msg === 'string'
 
-  const onSubmit = async (values: FieldValues) => {
-    try {
-      // @TODO: Replace this with API call when backend is ready
-      const { mnemonic } = await Promise.resolve({ mnemonic: DUMMY_MNEMONIC })
+  const isDecryptionError = (err: any): err is DecryptionError =>
+    err &&
+    typeof err.message === 'string' &&
```

Comment:
Since you're checking for defined `err.message` with the `?.` notation for properties access, you shouldn't need to check for `typeof`.

Reply:
Because I'm using startsWith(), if message is defined, but not a string, then startsWith will not be a function and it will throw.